modules/persistence/src/services/connector/pool/index.ts

```diff
@@ -0,0 +1,17 @@
+import { MongoClient, Db } from 'mongodb';
+
+// cached db object, so we can handle initial connection process once if unitialized
+let dbInstance: Db;
+// Handles memoization of db object, and initial connection logic if needs to be initialized
+export const db = async (client: MongoClient) => {
```

Comment:
I think that in a larger application or monolith I would have gone down that route, but since this is mostly a bounded context I've taken the approach of adding additional comments where it's exported in the higher level `connector/index.ts`.

The intent is that we shouldn't really have to make modifications in this file, so I prefer this approach - I'd rather hide the connection implementation/helpers here, and just export the `pool` connection from the `connector` with comment caveats around not touching it unless absolutely necessary.